core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1005,6 +1005,7 @@ public CompletionStats completionStats(String... fields) {
         }
         final long time = System.nanoTime();
         final Engine.CommitId commitId = engine.flush(force, waitIfOngoing);
+        engine.refresh("flush"); // TODO this is technically wrong we should remove this in 7.0
```

Comment:
OK. Now I see why the engine changes can be done in 6.x. I would opt for keeping as it was (i.e. full refresh on flush) and do as a small following in 7.0 to change the refresh command in the engine + mark it as breaking etc. Just a suggestion.